data/flexibleForms/forms.types.ts

```diff
@@ -69,6 +69,9 @@ export interface Form {
   tags?: string[];
   /** override the automatically generated /submissions/new... url, for edge case forms */
   canonicalUrl?: (socialCareId: number) => string;
+  dateOfEvent?: {
+    associatedId: string;
```

Comment:
why an object, rather than directly referencing a key somewhere in formAnswers?

Reply:
I think this way is quite explicit making for readable code. Also means if we need anything other than the associated id we just add another property.